### DISCUSSION TITLE
Google.Protobuf.ByteString needs to have a constructor with 0 args or only optional args. When mapping to byte[]

### DISCUSSION THREAD

**Discussion by schhwork** [LuckyPennySoftware/AutoMapper#3599] [NA]
Im getting the exception in the title still. I found an old issue #1457 where the same exception was discussed and @jbogard confimed it was a bug or missing feature if i understood that correctly. Ive reproduced this in a spikeproject just to confirm nothing else was affecting it. It seems that the value of my sorceproperty don't matter if its null or not. If i do the conversion outside on automapper it works fine.

### Source/destination types

```csharp

public class Source
{
    public byte[] Bytes { get; set; }
}

public class Destination
{
    public ByteString Bytes { get; set; }
}
```

### Mapping configuration

```csharp
var mapperConfiguration = new MapperConfiguration(cfg =>
{
    cfg.CreateMap<Source, Destination>();
    cfg.CreateMap<byte[], ByteString>().ConstructUsing(x => ByteArrayToByteString(x));
});

private ByteString ByteArrayToByteString(byte[] arg)
{
    return arg != null ? ByteString.CopyFrom(arg) : null;
}
```

### Expected behavior

ByteString mapped with contents of byte[]

### Actual behavior

Google.Protobuf.ByteString needs to have a constructor with 0 args or only optional args. 

### Steps to reproduce

```csharp
var source = new Source(); 

mapper.Map<Destination>(source);
```

**Comment by jbogard** [comment]
You should be using `ConvertUsing` here, not `ConstructUsing`. `ConstructUsing` means "construct, then map" where `ConvertUsing` means "just use this to map and ignore anything else".

**Reply by schhwork** [reply]
Thast makes perfect sense actually. ConvertUsing was what i expected. Thank you, i took it from one of our other mappings and didnt realize this. My mistake, but thank you for giving me an answer even though this was an oversight be me.
